### PR TITLE
Updated vSphere support matrix details

### PR DESCRIPTION
**Pull request by mylesagray**
Updated the vSphere support matrix to include details about the currently supported storage integration (in-tree VCP driver) and it's supported vSphere versions.

CSI support to be added later.

### CODE REVIEW

File: modules/installation-vsphere-infrastructure.adoc [hunk 1]
```diff
@@ -5,22 +5,21 @@
...
 = VMware vSphere infrastructure requirements
 
-You must install the {product-title} cluster on a VMware vSphere version 6.5U2
+You must install the {product-title} cluster on a VMware vSphere version 6.5
 or 6.7U2 or later instance.
 
-VMware recommends using vSphere Version 6.7U2 or later with your {product-title}
+VMware recommends using vSphere Version 6.5 or later with your {product-title}
 cluster.
 vSphere 6.7U2 includes:
```
Comment (kalexand-rh) [inline modules/installation-vsphere-infrastructure.adoc]: @mylesagray, should this be 6.5 as well?

@katherinedube, will you PTAL?

Comment (katherinedube) [reply modules/installation-vsphere-infrastructure.adoc]: 6.7U2 would still be a requirement for NSX-T SDN even if you remove vSAN CSI. So I think it should probably stay as 6.7U2. I'll let Myles confirm though.

Comment (mylesagray) [reply modules/installation-vsphere-infrastructure.adoc]: Agree with @katherinedube - Let's leave it at 6.7 U2 but remove CSI reference.

**Comment by kalexand-rh** [comment]
@mylesagray, will you squash your commits?

**Comment by mylesagray** [comment]
@kalexand-rh - squashed

**Pull request merged**